modules/dailymotionBidAdapter.js

```diff
@@ -0,0 +1,112 @@
+import { config } from '../src/config.js';
+import { registerBidder } from '../src/adapters/bidderFactory.js';
+import { VIDEO } from '../src/mediaTypes.js';
+import { deepAccess } from '../src/utils.js';
+
+/**
+ * Get video metadata from bid request
+ *
+ * @param {BidRequest} bidRequest A valid bid requests that should be sent to the Server.
+ * @return video metadata
+ */
+function getVideoMetadata(bidRequest) {
+  const videoAdUnit = deepAccess(bidRequest, 'mediaTypes.video', {});
+  const videoBidderParams = deepAccess(bidRequest, 'params.video', {});
+
+  const videoParams = {
+    ...videoAdUnit,
+    ...videoBidderParams, // Bidder Specific overrides
+  };
+
+  const videoMetadata = {
+    description: videoParams.description || '',
+    duration: videoParams.duration || 0,
+    iabcat2: Array.isArray(videoParams.iabcat2) ? videoParams.iabcat2 : [],
+    id: videoParams.id || '',
+    lang: videoParams.lang || '',
+    private: videoParams.private || false,
+    tags: videoParams.tags || '',
+    title: videoParams.title || '',
+    topics: videoParams.topics || '',
+    xid: videoParams.xid || '',
+  };
+
+  return videoMetadata;
+}
+
+export const spec = {
+  code: 'dailymotion',
+  gvlid: 573,
+  supportedMediaTypes: [VIDEO],
+
+  /**
+   * Determines whether or not the given bid request is valid.
+   * The only mandatory parameter for a bid to be valid is the API key.
+   * Other parameters are optional.
+   *
+   * @return boolean True if this is a valid bid, and false otherwise.
+   */
+  isBidRequestValid: function (bid) {
+    return ((typeof bid !== 'undefined') && (typeof bid?.params?.apiKey === 'string') && (bid.params.apiKey.length > 10));
+  },
+
+  /**
+   * Make a server request from the list of valid BidRequests (that already passed the isBidRequestValid call)
+   *
+   * @param {BidRequest[]} validBidRequests A non-empty list of valid bid requests that should be sent to the Server.
+   * @param {BidderRequest} bidderRequest
+   * @return ServerRequest Info describing the request to the server.
+   */
+  buildRequests: (validBidRequests = [], bidderRequest) => validBidRequests.map(bid => ({
+    method: 'POST',
+    url: 'https://pb.dmxleo.com',
+    data: {
+      bidder_request: {
+        gdprConsent: {
+          apiVersion: bidderRequest?.gdprConsent?.apiVersion || 1,
+          consentString: bidderRequest?.gdprConsent?.consentString || '',
+          // Cast boolean in any case (eg: if value is int) to ensure type
+          gdprApplies: !!bidderRequest?.gdprConsent?.gdprApplies,
+        },
+        refererInfo: {
+          page: bidderRequest?.refererInfo?.page || '',
+        },
+        uspConsent: bidderRequest?.uspConsent || '',
+      },
+      config: {
+        api_key: bid.params.apiKey
+      },
+      coppa: config.getConfig('coppa'),
+      request: {
```

Comment:
This request is quite sparse in comparison to your competitors. Why don't you want the publisher floor, the eids, or the user.data object? You also might find site.content.id will work for you as up to three vendors may be trying to set it in a race and you need the site.content ecids.